kernel/poll.c

```diff
@@ -470,11 +471,14 @@ static int signal_poll_event(struct k_poll_event *event, uint32_t state)
 void z_handle_obj_poll_events(sys_dlist_t *events, uint32_t state)
 {
 	struct k_poll_event *poll_event;
+	k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
Definitely looks like a lock was missing here, but I do worry about the possibility of a deadlock underneath signal_poll_event, which calls into a bunch of subsystems.  Are we 100% sure those can't turn around and call back into something that wants the poll lock?